examples/config/read-json.cpp

```diff
@@ -0,0 +1,87 @@
+// Illustrates how to read custom data types from JSON files.
+
+#include "caf/actor_system.hpp"
+#include "caf/actor_system_config.hpp"
+#include "caf/caf_main.hpp"
+#include "caf/json_reader.hpp"
+#include "caf/type_id.hpp"
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <string_view>
+
+constexpr std::string_view example_input = R"([
+  {
+    "id": 1,
+    "name": "John Doe"
+  },
+  {
+    "id": 2,
+    "name": "Jane Doe",
+    "email": "jane@doe.com"
+  }
+])";
+
+struct user {
+  uint32_t id;
+  std::string name;
+  std::optional<std::string> email;
+};
+
+template <class Inspector>
+bool inspect(Inspector& f, user& x) {
+  return f.object(x).fields(f.field("id", x.id), f.field("name", x.name),
+                            f.field("email", x.email));
+}
+
+using user_list = std::vector<user>;
+
+CAF_BEGIN_TYPE_ID_BLOCK(example_app, caf::first_custom_type_id)
+
+  CAF_ADD_TYPE_ID(example_app, (user))
+  CAF_ADD_TYPE_ID(example_app, (user_list))
+
+CAF_END_TYPE_ID_BLOCK(example_app)
+
+int caf_main(caf::actor_system& sys) {
+  // Get file path from config (positional argument).
+  auto& cfg = sys.config();
+  if (cfg.remainder.size() != 1) {
+    std::cerr << "*** expected one positional argument: path to a JSON file\n";
+    return EXIT_FAILURE;
+  }
```

Comment:
Maybe we should add support for that to our argument parser at some time. 🤔

Reply:
You mean positional arguments? Yeah, would be a good addition especially since the positional argument won't show up in the helptext at the moment.

Reply:
Exactly.